app/actions/tournaments.ts

```diff
@@ -44,7 +44,13 @@ export async function tournamentSubmissionFormAction(
           
           // Try to extract the id using regex
           const match = MatchLinkPattern.exec(s);
-          return match ? parseFloat(match[1]) : s;
+
+          if (match == null) {
+            return "<Invalid match link>";
+          }
+
+          const url = match[0];
+          return match ? parseFloat(url.substring(url.lastIndexOf('/') + 1)) : s;
```

Comment:
```suggestion
  return MatchLinkPattern.test(s) 
    ? parseFloat(s.substring(s.lastIndexOf('/') + 1)) 
    : s;
```

Reply:
Returning `"<Invalid match link>"` will break the error feedback on the front end. If we cannot parse the input into an int we should be returning the original string as that is what will display as the error. The reason why it shows `NaN` right now is because this:
```ts
const match = MatchLinkPattern.exec(s);
return match ? parseFloat(match[1]) : s;
```
Means the input is matching the regex, it just isn't parsing the correct matching group `match[1]` isn't actually the group containing the id